### ISSUE TITLE
Implement Cloud block

### ISSUE THREAD

**Issue by jderieg** [AzBuilder/terrakube#400]
### Feature description 💡

Terraform's preferred backend block is the Cloud block as of Terraform 1.1.0. It would be great to see this backend type implemented in Terrakube.
Along with that, the cloud block allows use of workspace tags, which allow for more dynamic workspace selection within Terraform code. These would be a great addition to the workspace settings in Terrakube.

Reference:
https://developer.hashicorp.com/terraform/language/settings/backends/remote
https://developer.hashicorp.com/terraform/cli/cloud/settings

**Comment by alfespa17** [comment]
Support to use terraform init with cloud block using workspace name was added in #406 

Support for terraform plan, apply and destroy will be added in other PRs

**Comment by jcanizalez** [comment]
Hello @alfespa17 @jderieg, this issue is now focused on implementing the cloud block. For the Workspace tagging feature, please see the new issue I created here: https://github.com/AzBuilder/terrakube/issues/409

I will try to add Workspace tagging in the next 2.14.0 version

**Comment by alfespa17** [comment]
Cloud block was implemented in #411 and #406 It will support the following usage:

```terraform
terraform {
  cloud {
    organization = "simple"
    hostname = "8080-azbuilder-terrakube-sscrnu9jbie.ws-us99.gitpod.io"

    workspaces {
      name = "samplecloud"
    }
  }
}
```

We are still missing the use of "tags", we will need to implement #409 before adding the support in the cloud block.

**Comment by alfespa17** [comment]
Support to use cloud block with tags will be added in #420, I am closing this issue because the basic cloud block usage is now supported in version 2.14.0
